app/controllers/microposts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class MicropostsController < ApplicationController
   before_action :logged_in_user, only: [:create, :destroy]
   before_action :correct_user, only: :destroy
-
+  rescue_from ::ActiveRecord::RecordNotFound, with: :record_not_found
```

Comment:
make new ApiController < ApplicationController
move it there with a new `private` method

Reply:
done